apps/src/codebridge/FileBrowser/prompts/openNewFilePrompt.tsx

```diff
@@ -0,0 +1,95 @@
+import {getNextFileId} from '@codebridge/codebridgeContext';
+import {NewFileFunction} from '@codebridge/codebridgeContext/types';
+import {DEFAULT_FOLDER_ID} from '@codebridge/constants';
+import {ProjectType, FolderId, ProjectFile} from '@codebridge/types';
+import {checkForDuplicateFilename, isValidFileName} from '@codebridge/utils';
+
+import codebridgeI18n from '@cdo/apps/codebridge/locale';
+import {
+  DialogType,
+  DialogControlInterface,
+  extractDialogClosePromiseInput,
+} from '@cdo/apps/lab2/views/dialogs';
+import {EVENTS} from '@cdo/apps/metrics/AnalyticsConstants';
+
+type ValidateNewFileNameArgs = {
+  fileName: string;
+  folderId: FolderId;
+  projectFiles: ProjectType['files'];
+  isStartMode: boolean;
+};
+
+const validateNewFileName = ({
+  fileName,
+  folderId,
+  projectFiles,
+  isStartMode,
+}: ValidateNewFileNameArgs) => {
+  if (!fileName.length) {
+    return;
+  }
+  if (!isValidFileName(fileName)) {
+    return codebridgeI18n.invalidNameError();
+  }
+  const duplicate = checkForDuplicateFilename({
+    fileName,
+    folderId,
+    projectFiles,
+    isStartMode,
+  });
+  if (duplicate) {
+    return duplicate;
+  }
+  const [, extension] = fileName.split('.');
+  if (!extension) {
+    return codebridgeI18n.noFileExtensionError();
+  }
+};
+
+type OpenNewFilePromptArgsType = {
+  folderId: FolderId;
+  appName: string;
+  dialogControl: Pick<DialogControlInterface, 'showDialog'>;
+  newFile: NewFileFunction;
+  projectFiles: ProjectType['files'];
+  sendCodebridgeAnalyticsEvent: (eventName: string) => unknown;
+  isStartMode: boolean;
+  validationFile: ProjectFile | undefined;
+};
+
+export const openNewFilePrompt = async ({
+  folderId = DEFAULT_FOLDER_ID,
+  appName,
+  dialogControl,
+  newFile,
+  projectFiles,
+  sendCodebridgeAnalyticsEvent,
+  isStartMode,
+  validationFile,
+}: OpenNewFilePromptArgsType) => {
+  const results = await dialogControl?.showDialog({
+    type: DialogType.GenericPrompt,
+    title: codebridgeI18n.newFilePrompt(),
+    validateInput: (fileName: string) =>
+      validateNewFileName({fileName, folderId, projectFiles, isStartMode}),
+  });
+  if (results.type !== 'confirm') {
+    return;
+  }
+  const fileName = extractDialogClosePromiseInput(results);
+
+  const files = Object.values(projectFiles);
+  // The validation file is in the project files in start mode.
+  if (validationFile && !isStartMode) {
+    files.push(validationFile);
+  }
+  const fileId = getNextFileId(files);
+
+  newFile({
+    fileId,
+    fileName,
+    folderId,
+  });
+
+  sendCodebridgeAnalyticsEvent(EVENTS.CODEBRIDGE_NEW_FILE);
+};
```

Comment:
and this separate function is what was embedded in `FileBrowser`, with minor tweaking to args